### PR TITLE
operator: add built-in 'Random' label templates

### LINKED ISSUE #802: [feature] Elemental Operator Labels: add $RANDOM labels to allow inserting random values in templated labels
That could be useful to allow for instance to have dynamic machineName values with a custom and fixed prefix / suffix

### PR DESCRIPTION
**Pull request by fgiudici**
introduce built-in random templates:
* ${Random/UUID}
* ${Random/Hex/[0-32]}
* ${Random/Int/[MAXINT]}

examples:
${Random/UUID}		--> e512d5ca-a765-42f2-82b7-264f37ffb329 ${Random/Hex/12}	--> e512d5caa765
${Random/Int/1000}	--> 199

Fixes #802

### CODE REVIEW

File: pkg/templater/random.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templater
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math/rand/v2"
```
Comment (anmazzotti) [inline pkg/templater/random.go:22]: Would it be possible to use `crypto/rand` instead? Maybe a bit overkill, but doesn't harm I think.

File: pkg/templater/templater.go [hunk 2]
```diff
@@ -72,14 +73,32 @@ func replaceStringData(data map[string]interface{}, name string) (string, error)
...
 		result.WriteString(str[:i])
-		obj := values.GetValueN(data, strings.Split(str[i+2:j+i], "/")...)
-		if str, ok := obj.(string); ok {
-			result.WriteString(str)
-		} else {
-			return "", errValueNotFound
+		tmplVal := strings.Split(str[i+2:j+i], "/")
+
+		strVal, err := templateToString(data, tmplVal)
+		fmt.Printf("TMPL CONVERSION: %q --> %q\n", tmplVal, strVal)
```
Comment (anmazzotti) [inline pkg/templater/templater.go:79]: Debug log maybe? :P

**Review by anmazzotti — APPROVED** [review]
Looks good with a couple of minor comments. 

I also wonder if this should replace the current behavior of `m-$UUID` or is that already part of the PR?

File: pkg/templater/templater.go (hunk 2 above)
Comment (fgiudici) [reply pkg/templater/templater.go:79]: 🤦🏼  good catch, left overs from initial development!

File: pkg/templater/random.go (hunk 1 above)
Comment (fgiudici) [reply pkg/templater/random.go:22]: In this case rand/v2 would be enough: it introduced dynamic seeding and reworked the random number generator algs wrt rand/v1.  But well, better safe than sorry, and if we always stick to crypto/rand we will not "forget" to use it where it could be really needed in the project.

Comment (fgiudici) [reply pkg/templater/random.go]: Well, honestly I'm not that happy with crypto/rand here: it is a bit cumbersome generating simple random INTs and we don't really need "secure" numbers. Mainly the "big".Int thing requirement is annoying (while of course it makes a lot of sense in security contexts).
Here it is just a minimal rand usage, so doesn't really matter what library we use... sticking with crypto/rand.
But if we will ever need more random operations in the projects with no requirements from a security PoV let's stick with math/rand/v2.
So, long story short, changed my mind about "always" sticking with crypto/rand in the project if security is not involved 😆

**Comment by fgiudici** [comment]
> Looks good with a couple of minor comments.
> 
> I also wonder if this should replace the current behavior of `m-$UUID` or is that already part of the PR?

Well,  `m-$UUID` is still the default `machineName`, but yes, we can now generate custom machine names, e.g.:
`machineName: cattle${Random/Int/5}-${Random/Hex/8}`
that would generate hostnames (and machineinventory names) with names like:
`cattle0-02401844`
`cattle3-1823bf5a`
